File: src/main/java/sk/hudak/prco/manager/AddingNewProductManager.java
package sk.hudak.prco.manager;

import sk.hudak.prco.api.EshopUuid;

public interface AddingNewProductManager {

    /**
     * @param productsUrl list of new product URL's
     */
    void addNewProductsByUrl(String... productsUrl);

    /**
     * Vyhlada produkty s danym klucovym slovom pre konkretny eshop a ulozi ich do tabulky NEW_PRODUCT.
     *
     * @param eshopUuid     eshop identifikator
     * @param searchKeyWord search key word
     */
    void addNewProductsByKeywordForEshop(EshopUuid eshopUuid, String searchKeyWord);

    /**
     * Vyhlada produkty s danym klucovym slovom pre vsetky eshopy a ulozi ich do tabulky NEW_PRODUCT.
     *
     * @param searchKeyWord search key word
     */
    void addNewProductsByKeywordForAllEshops(String searchKeyWord);

    /**
     * @param searchKeyWords search key words
     */
    void addNewProductsByKeywordsForAllEshops(String... searchKeyWords);
}

File: src/main/java/sk/hudak/prco/manager/addprocess/AddingNewProductManager.kt
package sk.hudak.prco.manager.addprocess

import sk.hudak.prco.api.EshopUuid

interface AddingNewProductManager {

    /**
     * @param productsUrl list of new product URL's
     */
    fun addNewProductsByUrl(productsUrl: List<String>)

    /**
     * Vyhlada produkty s danym klucovym slovom pre konkretny eshop a ulozi ich do tabulky NEW_PRODUCT.
     *
     * @param eshopUuid     eshop identifikator
     * @param searchKeyWord search key word
     */
    fun addNewProductsByKeywordForEshop(eshopUuid: EshopUuid, searchKeyWord: String)

    /**
     * Vyhlada produkty s danym klucovym slovom pre vsetky eshopy a ulozi ich do tabulky NEW_PRODUCT.
     *
     * @param searchKeyWord search key word
     */
    fun addNewProductsByKeywordForAllEshops(searchKeyWord: String)

    /**
     * @param searchKeyWords search key words
     */
    fun addNewProductsByKeywordsForAllEshops(vararg searchKeyWords: String)
}
